collect_app/src/main/java/org/fieldsight/naxa/site/SiteListFragment.java

```diff
@@ -130,6 +148,24 @@ private void changeActionVisibility(boolean visible) {
             return;
         }
         sortActionFilter.setVisible(visible);
+    }
+
+    private TermsLabels getTermsAndLabels() {
+        if(loadedProject == null) {
+            return null;
+        }
+        if(!TextUtils.isEmpty(loadedProject.getTerms_and_labels())) {
+            try{
+                Timber.i("ProjectDashBoardActivity:: terms and labels = %s", loadedProject.getTerms_and_labels());
+                JSONObject tlJson = new JSONObject(loadedProject.getTerms_and_labels());
+                return TermsLabels.fromJSON(tlJson);
+            }catch (Exception e){
+                e.printStackTrace();
```

Comment:
Could there be any case where `getTerms_and_labels()` could return null or a improper json ?